src/react-queries/queries/common.ts

```diff
@@ -90,43 +89,15 @@ export const common = createQueryKeys('common', {
   }),
   evmFees: ({
     data,
-    wallet,
     accountNumber,
     to,
-    from,
     value,
-    feeAsset,
-    feeAssetMarketData,
-  }: {
-    to: string
-    // Only used to make the queryKey unique by from address, since wallet doesn't serialize and is excluded from the queryKey
-    from: string
-    accountNumber: number
-    wallet: HDWallet
-    data: string
-    value: string
-    feeAsset: Asset
-    feeAssetMarketData: MarketData
+    chainId,
+  }: Omit<GetFeesWithWalletArgs, 'wallet' | 'adapter' | 'accountNumber' | 'data'> & {
+    accountNumber: number | undefined
+    data: string | undefined
+    chainId: ChainId
```

Comment:
Not blocking at all, but 💡 this could benefit from a helper type to make specific fields partial:

```
type PartialFields<T, K extends keyof T> = Omit<T, K> & Partial<Pick<T, K>>
```

So then you could do this to ensure the original non-optional types are preserved:

```
OptionalFields<Omit<GetFeesWithWalletArgs, 'wallet' | 'adapter'> & { chainId: ChainId }, 'accountNumber' | data'>
```

Reply:
Big 🧠, was precisely thinking of such util but my I disregarded it as "I don't have a type-system black belt can't do that with my TS fu", will add!